src/axom/spin/internal/linear_bvh/TraversalPredicates.hpp

```diff
@@ -179,6 +274,64 @@ class TraversalPredicates< DIMENSION_3, FloatType >
     return in_right;
   }
 
+  template < typename RayType >
+  AXOM_HOST_DEVICE
+  static inline bool rayIntersectsLeftBin(
+      const RayType& r, const vec4_t& s1, const vec4_t& s2,
+      FloatType TOL=numerics::floating_point_limits<FloatType>::epsilon()
+      ) noexcept
+  {
+    const FloatType& x0 = r[ 0 ];
+    const FloatType& y0 = r[ 1 ];
+    const FloatType& z0 = r[ 2 ];
+    const FloatType& nx = r[ 3 ];
+    const FloatType& ny = r[ 4 ];
+    const FloatType& nz = r[ 5 ];
+
+    // extract left bin, see BVHData.hpp for the internal BVH layout
+    const FloatType& xmin = s1[ 0 ];
```

Comment:
Similarly to my comment in the 2D case -- 
Perhaps a utility function in ``BVHData.hpp`` could be used to hide the implementation details?

Reply:
Ditto.